setup.py

```diff
@@ -1,96 +1,25 @@
-    which is now recommended by setuptools. It first appeared in 36.2.0, went broken with 36.2.1, and
-    again worked since 36.2.2, so we're using that. See:
-    https://setuptools.readthedocs.io/en/latest/history.html#v36-2-2
-    https://github.com/pypa/setuptools/issues/1099
-
-    References:
-
-    * https://wheel.readthedocs.io/en/latest/index.html#defining-conditional-dependencies
-    * https://www.python.org/dev/peps/pep-0426/#environment-markers
-    * https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-platform-specific-dependencies
-    """
-    try:
-        version = pkg_resources.parse_version(setuptools.__version__)
-        if version >= pkg_resources.parse_version("36.2.2"):
-            return 2
-        if version >= pkg_resources.parse_version("0.7.2"):
-            return 1
-    except Exception as exc:
-        sys.stderr.write("Could not test setuptool's version: %s\n" % exc)
-
-    # as of testing on 2018-05-26 fedora was on version 37* and debian was on version 33+
-    # we should consider erroring on those
-    return 0
-
 
 def main():
-    extras_require = {}
     install_requires = [
         "py>=1.5.0",  # if py gets upgrade to >=1.6, remove _width_of_current_line in terminal.py
         "six>=1.10.0",
         "setuptools",
         "attrs>=17.4.0",
         "more-itertools>=4.0.0",
         "atomicwrites>=1.0",
+        'funcsigs;python_version<"3.0"',
+        'pathlib2>=2.2.0;python_version<"3.6"',
+        'colorama;sys_platform=="win32"',
     ]
     # if _PYTEST_SETUP_SKIP_PLUGGY_DEP is set, skip installing pluggy;
     # used by tox.ini to test with pluggy master
     if "_PYTEST_SETUP_SKIP_PLUGGY_DEP" not in os.environ:
         install_requires.append("pluggy>=0.7")
-    environment_marker_support_level = get_environment_marker_support_level()
-    if environment_marker_support_level >= 2:
-        install_requires.append('funcsigs;python_version<"3.0"')
-        install_requires.append('pathlib2>=2.2.0;python_version<"3.6"')
-        install_requires.append('colorama;sys_platform=="win32"')
-    elif environment_marker_support_level == 1:
-        extras_require[':python_version<"3.0"'] = ["funcsigs"]
-        extras_require[':python_version<"3.6"'] = ["pathlib2>=2.2.0"]
-        extras_require[':sys_platform=="win32"'] = ["colorama"]
-    else:
-        if sys.platform == "win32":
-            install_requires.append("colorama")
-        if sys.version_info < (3, 0):
-            install_requires.append("funcsigs")
-        if sys.version_info < (3, 6):
-            install_requires.append("pathlib2>=2.2.0")
 
     setup(
-        name="pytest",
-        description="pytest: simple powerful testing with Python",
-        long_description=long_description,
         use_scm_version={"write_to": "src/_pytest/_version.py"},
         url="https://docs.pytest.org/en/latest/",
         project_urls={
```

Comment:
Any other metadata that we can add to `setup.cfg`? Like `license`, `platforms`, `author`, `entry_points`, ...?

Reply:
yes